internal/app/dogma/get_cmd.go

```diff
@@ -66,6 +78,139 @@ func (gf *getFileCommand) execute(c *cli.Context) error {
 	return nil
 }
 
+type getDirectoryCommand struct {
+	out           io.Writer
+	repo          repositoryRequestInfo
+	localFilePath string
+}
+
+func (gd *getDirectoryCommand) execute(c *cli.Context) error {
+	client, err := newDogmaClient(c, gd.repo.remoteURL)
+	if err != nil {
+		return err
+	}
+
+	return gd.executeWithDogmaClient(c, client)
+}
+
+func (gd *getDirectoryCommand) executeWithDogmaClient(_ *cli.Context, client *centraldogma.Client) error {
+	repo := gd.repo
+	entry, err := getRemoteFileEntryWithDogmaClient(client,
+		repo.projName, repo.repoName, repo.path, repo.revision, nil)
+	if err != nil {
+		return err
+	}
```

Comment:
I found a bug that occurred when a path is a root path (`/`).
> json: cannot unmarshal array into Go value of type struct { Type string "json:\"type\""; *centraldogma.Alias }

This happens because Central Dogma returns a list of entries for `/`.
How about assuming an entry type for `/` as a directory and calling `ListFiles` without `getRemoteFileEntryWithDogmaClient?

Reply:
are you sure? isn't using `/` means it's a repository? Will it be better to return an error that said we can't download a whole repository instead?

Reply:
I don't think we need to prevent users from downloading the whole repository.
Additionally, if a glob pattern such as `myProj/myRepo/a/path/**` is used, we will get the same error. On second thought, it would be better to handle multiple entries returned by the servers.
```json
[
  {
    "revision": xx,
    "path": "/abc",
    "type": "DIRECTORY",
    "url": "/api/v1/projects/ikhun_test/repos/hello1/contents/abc"
  },
  {
    "revision": xx,
    "path": "/foo.json",
    "type": "JSON",
    "content": {
      "foo": "1"
    },
    "url": "/api/v1/projects/ikhun_test/repos/hello1/contents/foo.json"
  },
  {
    "revision": xx,
    "path": "/def.txt",
    "type": "TEXT",
    "content": "hello world",
    "url": "/api/v1/projects/ikhun_test/repos/hello1/contents/txt"
  }
]
```

Reply:
sure, Let's merge this then I'll create a new PR for this improvement, wdyt?

Reply:
Sounds good. 👍